src/streamlink/stream/hls_playlist.py

```diff
@@ -251,6 +254,7 @@ def parse(self, data):
             return self.m3u8
         else:
             if not line.startswith("#EXTM3U"):
+                log.debug("First line of playlist was {0}", line)
```

Comment:
I'd actually make it a `warning` log message, or maybe even an `error` log message.
```suggestion
                log.warning("Malformed HLS Playlist. Expected #EXTM3U, but got {0}".format(line))
```